File: Java-Core/src/main/java/presto/type/VoidType.java
package presto.type;

import presto.runtime.Context;


public class VoidType extends NativeType {

	static VoidType instance = new VoidType();
	
	public static VoidType instance() {
		return instance;
	}
	
	private VoidType() {
		super("Void");
	}
	
	@Override
	public Class<?> toJavaClass() {
		return void.class;
	}

	@Override
	public boolean isAssignableTo(Context context, IType other) {
		throw new RuntimeException("Should never get there !");
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/type/VoidType.js
var NativeType = require("./NativeType").NativeType;


function VoidType() {
	NativeType.call(this, "Void");
	return this;
}

VoidType.prototype = Object.create(NativeType.prototype);
VoidType.prototype.constructor = VoidType;

VoidType.instance = new VoidType();
	
/*	
	@Override
	public Class<?> toJavaClass() {
		return void.class;
	}

	@Override
	public boolean isAssignableTo(Context context, IType other) {
		throw new RuntimeException("Should never get there !");
	}
	
*/

exports.VoidType = VoidType;
